pyensign/ensign.py

```diff
@@ -68,22 +69,24 @@ def __init__(
         creds = {"client_id": client_id, "client_secret": client_secret}
         auth = AuthClient(auth_url, creds)
         connection = Connection(addrport=endpoint, insecure=insecure, auth=auth)
-        self.client = Client(connection)
 
         if disable_topic_cache:
             self.topics = None
         else:
             self.topics = Cache()
 
-    async def publish(self, topic_name, *events, on_ack=None, on_nack=None):
+        self.client = Client(connection, topic_cache=self.topics)
+
+    async def publish(
+        self, topic, *events, on_ack=None, on_nack=None, ensure_exists=False
+    ):
         """
-        Publish events to an Ensign topic. If the topic doesn't exist it will be
-        created.
+        Publish events to an Ensign topic.
 
         Parameters
         ----------
-        topic_name : str
-            The name of the topic to publish events to.
+        topic: str
+            The name or ID of the topic to publish events to.
```

Comment:
In Ensign office hours there was some discussion about if we should allow both topic IDs and topic names on publish/subscribe, and when to do existence checks. Here's what we decided:

1. On publish and subscribe, both topic IDs and topic names are allowed. If the topic name is in the cache then lookup the ID using the cache, otherwise try to parse the topic ID from the string. An error should be returned if pyensign can't resolve the topic into a topic ID.
2. Publish has an optional parameter to create the topic if it doesn't exist, but it's `false` by default. Topic creation should happen in a separate API call by the user.
3. Subscribe should not create any topics, it should just make the subscribe request to the server with the topic IDs and the Ensign server is responsible for returning errors if the topics don't exist.

Reply:
Thanks for the explanation!